graylog2-web-interface/src/views/components/actions/ActionDropdown.tsx

```diff
@@ -0,0 +1,129 @@
+
+import { MenuItem } from 'components/graylog';
+import ActionMenuItem from 'views/components/actions/ActionMenuItem';
+import FieldAndValueActionsContext from 'views/components/contexts/FieldAndValueActionsContext';
+import { ActionDefinition, ActionHandlerArguments } from 'views/components/actions/ActionHandler';
+
+const DropdownHeader = styled.span`
+  padding-left: 10px;
+  padding-right: 10px;
+  padding-bottom: 5px;
+  margin-bottom: 5px;
+  font-weight: 600;
+`;
+
+const StyledListItem = styled.li`
+  margin-bottom: 10px;
+  list-style: none;
+`;
+
+const filterVisibleActions = (actions: Array<ActionDefinition> | undefined = [], handlerArgs: Props['handlerArgs']) => {
+  return actions?.filter((action: ActionDefinition) => {
+    const { isHidden = () => false } = action;
+
+    return !isHidden(handlerArgs);
+  });
+};
+
+const useInternalActions = (type: Props['type'], handlerArgs: Props['handlerArgs']) => {
+  const fieldAndValueActions = useContext(FieldAndValueActionsContext);
+
+  if (type === 'value' && fieldAndValueActions) {
+    return filterVisibleActions(fieldAndValueActions.valueActions.internal, handlerArgs);
+  }
+
+  if (type === 'field' && fieldAndValueActions) {
+    return filterVisibleActions(fieldAndValueActions.fieldActions.internal, handlerArgs);
+  }
+
+  return [];
+};
+
+const useExternalActions = (type: Props['type'], handlerArgs: Props['handlerArgs']) => {
+  const fieldAndValueActions = useContext(FieldAndValueActionsContext);
+
+  if (type !== 'value' || !fieldAndValueActions) {
+    return [];
+  }
+
+  return filterVisibleActions(fieldAndValueActions.valueActions.external, handlerArgs);
+};
+
+type Props = {
+  children: React.ReactNode,
+  type: 'field' | 'value',
+  handlerArgs: ActionHandlerArguments,
+  setOverflowingComponents: (components: React.ReactNode) => void,
+  overflowingComponents: React.ReactNode,
+  onMenuToggle: () => void,
+};
+
+const ActionDropdown = ({
+  children,
+  type,
+  handlerArgs,
+  setOverflowingComponents,
+  overflowingComponents,
+  onMenuToggle,
+}: Props) => {
+  const internalActions = useInternalActions(type, handlerArgs);
+  const externalActions = useExternalActions(type, handlerArgs);
+
+  return (
+    <>
+      <StyledListItem>
+        <DropdownHeader>
+          {children}
+        </DropdownHeader>
+      </StyledListItem>
+
+      <MenuItem divider />
+      <MenuItem header>Actions</MenuItem>
+      {internalActions.map((action) => (
+        <ActionMenuItem action={action}
+                        key={`${type}-action-${action.type}`}
+                        handlerArgs={handlerArgs}
+                        setOverflowingComponents={setOverflowingComponents}
+                        overflowingComponents={overflowingComponents}
+                        type={type}
+                        onMenuToggle={onMenuToggle} />
+      ))}
+
+      {(externalActions && externalActions.length !== 0) && (
+        <>
+          <MenuItem divider />
```

Comment:
Please note, I remove the header for the second dropdown section, because neither "External Actions" nor "Event Actions" really described what these actions are doing. I kept the divider to still separate these external actions from the internal actions.

Reply:
External Actions are IMHO the correct term. The actions are executed on non views/search/dashboard features and are therefore external to that system. In all but one case for now they are even external to graylog.

But we can bring that case to product management or our UX engineer.